### PR TITLE
WIP: Add support for memberspace

### LINKED ISSUE #631: Add memberspace to Syna
**Is this a BUG REPORT or FEATURE REQUEST?**:
feature

**What happened**:
Currently creating membership sites isn't possible with Syna.

**What you expected to happen**:
Make membership sites possible with Syna.

**Anything else we need to know?**:
Not really sure how this would work or even if it works.
Wanted to note it down.

### PR DESCRIPTION
**Pull request by mpourismaiel**
**What this PR does / why we need it**:
Add support for memberspace

**Which issue this PR fixes**:
fixes #631 

**Special notes for your reviewer**:
The implementation works with a fragment that you can add to _global or any page or section globals and in turn nav fragment in those pages will display a login link. I didn't add anymore customizations since I'm not even sure if this is the best approach. My assumptions is that the user would create a memberspace.md file in _global with following content:

```
+++
fragment = "memberspace"
subdomain = "subdomain_copied_from_installation_guide_of_memberspace"
+++
```

And then the nav fragment will display a link to Login/Signup page in Memberspace. It would be recommended that the button Memberspace provides be disabled from their website. But we can also move the "go to top" button as we discussed.

If you have any other approach in mind please let me know and if everything is okay I can add documentation and we can move forward with the merge.

**Release note**:
```release-note
- memberspace: New fragment that adds memberspace scripts to enable the service for Syna
```

### CODE REVIEW

File: layouts/partials/helpers/fragments.html [hunk 1]
```diff
@@ -172,6 +172,7 @@
   looking from the page directory or file to it's parents and so on. */}}
 {{- $page_scratch.Set "page_fragments" (slice) -}}
 {{- $page_scratch.Set "page_config" (slice) -}}
+{{- $page_scratch.Set "memberspace" nil -}}
```
Comment (stp-ip) [inline layouts/partials/helpers/fragments.html:175]: Using a global variable for a vendor specific fragment seems problematic.

File: layouts/partials/fragments/nav.html [hunk 2]
```diff
@@ -81,6 +81,13 @@
               </a>
             </li>
           {{- end -}}
+        {{- end -}}
+        {{- with ($page_scratch.Get "memberspace") }}
```
Comment (stp-ip) [inline layouts/partials/fragments/nav.html:85]: Adding a vendor specific fragment into another fragments code seems problematic.
We should inject it from a memberspace fragment or at least make this injection more generic.
As it currently stands this seems too bundled.

**Comment by mpourismaiel** [comment]
Currently I can't think of any approach that would be less complex. I would love to hear ideas on the matter. Maybe remove the fragment and make it a site configuration, applied to the baseof template?

**Comment by JOduMonT** [comment]
I'm curious, it is like a paywall ?
What manage the authentication Discourse, Netlify ?

**Comment by stp-ip** [comment]
Memberspace is a paywall service especially for membership use cases. They have different modes of security. Simplest will remove content until payment via JS, one will add the content in via JS from their DB after payment/login.

Authentication is therefore done on the memberspace side of things.

**Comment by JOduMonT** [comment]
I didnt knew it was a SaaS
I wonder if these bypass works ?
- https://github.com/iamadamdev/bypass-paywalls-chrome
- https://github.com/iamadamdev/bypass-paywalls-firefox

**Comment by stp-ip** [comment]
Most likely depends on the mode used.
But that's not part of Syna so not really relevant here.

**Comment by stp-ip** [comment]
Let's shelf this for now. We first need a better idea around provider fragments and also can't add things globally for a fragment.

Possible option for injecting something like this is the fragments JS or our event framework. Let's discuss this separately.

**Pull request closed without merging**